File: mediq-frontend/src/services/campService.js
import api from './api';

export const campService = {
  createCamp: async (data) => api.post('/camps', data),
  updateCamp: async (id, data) => api.put(`/camps/${id}`, data),
  getCampById: async (id) => api.get(`/camps/${id}`),
  getCampByCode: async (campCode) => api.get(`/camps/code/${campCode}`),
  assignStaff: async (id, staffData) => api.post(`/camps/${id}/staff`, staffData),
  updateCampStatus: async (id, status) => api.patch(`/camps/${id}/status`, null, { params: { status } }),
  getCampsByStatus: async (status) => api.get(`/camps/status/${status}`),
  searchCamps: async (status, keyword, page = 0, size = 10) =>
    api.get('/camps', { params: { status, keyword, page, size } }),
  deleteCamp: async (id) => api.delete(`/camps/${id}`),
};

export default campService;

File: mediq-backend/src/main/java/com/mediq/service/CampService.java
package com.mediq.service;

import com.mediq.constants.CampStatus;
import com.mediq.dto.AssignStaffRequest;
import com.mediq.dto.CampResponse;
import com.mediq.dto.CreateCampRequest;
import com.mediq.dto.UpdateCampRequest;
import org.springframework.data.domain.Page;

import java.util.List;

public interface CampService {

    CampResponse createCamp(CreateCampRequest request);

    CampResponse updateCamp(Long campId, UpdateCampRequest request);

    CampResponse getCampById(Long campId);

    CampResponse getCampByCode(String campCode);

    CampResponse assignStaff(Long campId, AssignStaffRequest request);

    CampResponse updateCampStatus(Long campId, CampStatus status);

    List<CampResponse> getCampsByStatus(CampStatus status);

    Page<CampResponse> searchCamps(CampStatus status, String keyword, int page, int size, String sortBy, String sortDir);

    void deleteCamp(Long campId);
}
